### DISCUSSION TITLE
UI improvement: Mark functions in extension with an icon

### DISCUSSION THREAD

**Discussion by Wend1go** [4ian/GDevelop#3507] [Archive]
## Description

When working on a larger external extension it gets hard to find the function you are looking for in the list.
There should be a visual way to distinguish between conditions, actions and expressions.

## Solution suggested

Mockup:
![image](https://user-images.githubusercontent.com/7413059/145811692-351a4b42-15e4-4a89-8a2e-32e0127f5b6e.png)

**Describe the solution**
Adding an icon in front of each function name makes it easy to see if we are dealing with a condition, action or expression.

## Alternatives considered

One could choose a name prefix for functions like "actionDoThis" / "aDoThis" or "condition_IsThis" / "cIsThis" but since the function names are also later visible in the event sheet when they get used this would make the events harder to read.

**Comment by 4ian** [comment]
Totally, that's an easy win to have an icon. Also a good opportunity to add a "lock" or something like this to show private functions/methods.

**Comment by Bouh** [comment]
If you need some icons based on the function icon.

Function: ![function32](https://user-images.githubusercontent.com/1670670/145891241-0b262957-b474-4ffe-8033-9c78173b7665.png)


StringExpression: ![functionStringExpression](https://user-images.githubusercontent.com/1670670/145891212-ccfa0c13-0c54-40aa-adca-100ba37a0f0f.png)
Action: ![functionAction](https://user-images.githubusercontent.com/1670670/145891215-e5fa1bbe-0e92-4990-861c-c863cd02f726.png)
Condition: ![functionCondition](https://user-images.githubusercontent.com/1670670/145891216-c86e2d56-62e2-4b8d-9ba8-0cbfd13063c0.png)
Expression: ![functionExpression](https://user-images.githubusercontent.com/1670670/145891217-0ae0010c-6079-4f27-9a8f-5d88c4fffa53.png)

**Comment by Bouh** [comment]
A new couple of icons in SVG.

Action, condition, expression.


[icons_functions.zip](https://github.com/4ian/GDevelop/files/7728238/icons_functions.zip)



![image](https://user-images.githubusercontent.com/1670670/146393688-6bfbcfd5-27a5-4514-942b-642d90e66ac0.png)
